util.rb

```diff
@@ -0,0 +1,61 @@
+require 'sinatra/base'
+require_relative 'environment'
+require 'time'
+
+# Util is a class of utility methods
+class Util
+  def self.max_value(data)
+    # Return the maximum value field from a list of hashes
+    # an empty list should return 0
+    return 0 if data.empty?
+    max = data[0][:value]
+    data.each do |line|
+      max = line[:value] if line[:value] > max
+    end
+    max # return
+  end # maxValue
+
+  def self.min_value(data)
+    # Return the minimum Value field in a list of hashes
+    # an empty list should return 0
+    return 0 if data.empty?
+    min = data[0][:value]
+    data.each do |line|
+      min = line[:value] if line[:value] < min
+    end
+    min # return
+  end # minValue
+
+  def self.avg_value(data)
+    # Return the avg (mean) Value field in a list of hashes
+    # an empty list should return 0
+    return 0 if data.empty?
+    total = 0
+    data.each do |line|
+      total += line[:value]
+    end
+    total /= data.length
+  end # avgValue
+
+  def self.days_in_range(data)
+    # Return number of days in range of data
+    # an empty range includes 0 days
+    return 0 if data.empty?
+    earliest = data[0][:created]
+    latest = data[-1][:created]
+    now = Time.now() # dates cannot be in the future
+    data.each do |line|
+      l = line[:created]
+      # dates must be in a Time format
+      return nil if l.class != Time
+      # dates cannot be in the future
+      return nil if l > now
+      earliest = l if l < earliest
+      latest = l if l > latest
+    end
+    # latest-earliest is number of seconds between two dates
+    # there are 60*60*24 seconds every day (leap years? DST?)
+    # round up to whole days?  Maybe not?
+    ((latest - earliest) / (60 * 60 * 24)).ceil
```

Comment:
It's true that this answer won't cover leap years nor will it account for daylight savings times.  I'm not sure whether rounding matters.  However, if I had to calculate the number of days between two Time values, I'd probably look at http://stackoverflow.com/questions/9608116/number-of-days-between-two-time-instances -- and while the accepted answer is the rounded version of yours, I personally like the simpler version of the second answer.

The `Enumerable` mixin may also be helpful here to identify the earliest and latest times.